File: server/src/main/java/gg/mineral/server/network/packet/status/clientbound/ResponsePacket.java
package gg.mineral.server.network.packet.status.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record ResponsePacket(String jsonResponse) implements Packet.OUTGOING {

    @Override
    public void serialize(ByteBuf os) {
        writeString(os, jsonResponse);
    }

    @Override
    public byte getId() {
        return 0x00;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/status/clientbound/ResponsePacket.kt
package gg.mineral.server.network.packet.status.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

data class ResponsePacket(val jsonResponse: String) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) =
        os.writeString(jsonResponse)

    override val id: Byte
        get() = 0x00
}
